homeassistant/components/envertech_evt800/sensor.py

```diff
@@ -0,0 +1,233 @@
+    ),
+    "temperature_2": SensorEntityDescription(
+        key="temperature_2",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        suggested_display_precision=1,
+        translation_key="temperature_2",
+    ),
+    "total_energy_1": SensorEntityDescription(
+        key="total_energy_1",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+        device_class=SensorDeviceClass.ENERGY,
+        suggested_display_precision=2,
+        translation_key="total_energy_1",
+    ),
+    "total_energy_2": SensorEntityDescription(
+        key="total_energy_2",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+        device_class=SensorDeviceClass.ENERGY,
+        suggested_display_precision=2,
+        translation_key="total_energy_2",
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: EnvertechEVT800ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up Envertech EVT800 sensors."""
+
+    evt800: pyenvertechevt800.EnvertechEVT800 = config_entry.runtime_data.client
+    coordinator: EnvertechEVT800Coordinator = config_entry.runtime_data
+
+    entities = []
+    for name, description in SENSORS.items():
+        data = evt800.data.get(name)
+        entities.append(
+            EnvertechEVT800Sensor(
+                evt800,
+                coordinator,
+                config_entry,
+                description,
+                name,
+                data,
+            )
+        )
+    if entities:
+        async_add_entities(entities)
+
+
+class EnvertechEVT800Sensor(EnvertechEVT800Entity, SensorEntity):
+    """Representation of a Envertech EVT800 sensor."""
+
+    def __init__(
+        self,
+        evt800: pyenvertechevt800.EnvertechEVT800,
+        coordinator: EnvertechEVT800Coordinator,
+        config_entry: ConfigEntry[Any],
+        description: SensorEntityDescription | None,
+        name: str,
+        value: Any,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(evt800, coordinator, config_entry)
+        if description is not None:
+            self.entity_description = description
+        else:
+            self._attr_name = name
+
+        self._value = value
+        self._device = evt800
+        self._key = name
+
+        self._attr_has_entity_name = True
+        self._attr_unique_id = f"{config_entry.unique_id}-{name}"
+        self._attr_native_value = self._device.data[self._key]
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Return the state of the sensor."""
+        self._attr_native_value = self._device.data[self._key]
+        self.async_write_ha_state()
+        super()._handle_coordinator_update()
```

Comment:
Instead, you can implement the native_value property, then you don't have to declare this twice